nautobot/dcim/models/devices.py

```diff
@@ -1209,3 +1217,91 @@ def __str__(self):
         if self.alias:
             return self.alias
         return f"{self.platform} - {self.version}"
+
+
+#
+# Controller
+#
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "locations",
+    "statuses",
+    "webhooks",
+)
+class Controller(PrimaryModel, ConfigContextModel):
+    """Controller model."""  # TODO: (whitej6) finialize description
+
+    name = models.CharField(max_length=255, unique=True)
```

Comment:
Will need to update branch to pull in latest from `next` including #5355, then can update this:

```suggestion
    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
```